### ISSUE TITLE
[FR] Block context menu

### ISSUE THREAD

**Issue by ebroder** [NemesisRE/kiosk-mode#130]
**Is your feature request related to a problem? Please describe.**

I'm working on building a kiosk for home control, and I'd like to limit access beyond the specific elements I'm putting on a dashboard. With a touchscreen at least, Chromium seems to support opening a context menu by long-pressing on the page.

**Describe the solution you'd like**

Given that there's already a `block_mouse` option, I was wondering if you'd be open to a `block_contextmenu` option. It should be fairly easy to implement in Javascript with something like `document.addEventListener('contextmenu', e => e.preventDefault())`

**Describe alternatives you've considered**

I'm not aware of any other way to block the context menu, short of introducing my own JS module.

**Comment by elchininet** [comment]
Hi @ebroder,
It doesn‘t happen in Chrome on my side. What appears in this context menu? It is some Home Assistant menu or a native menu from the browser?
Regards

**Comment by ebroder** [comment]
It's the native browser menu, but you can block it with JS. Here's a video of what I'm seeing:

https://github.com/NemesisRE/kiosk-mode/assets/28167/7108e3f1-c2fb-4e28-b277-1e7209f7725a

**Comment by elchininet** [comment]
Hi @ebroder,
[It seems indeed a Chromium issue](https://github.com/jareware/chilipie-kiosk/issues/64). I know that the `contextmenu` could be blocked with `JavaScript` and this could be included in `kiosk-mode` but did you test it already on your specific device? Some users reported [that it didn‘t work](https://github.com/jareware/chilipie-kiosk/issues/64#issuecomment-1759250665) in their devices. (and talking about Chromium we already know that some things that work in the regular browsers [don‘t work in that browser](https://github.com/NemesisRE/kiosk-mode/issues/102)).
Regards.

**Comment by elchininet** [comment]
It would be good to test if `user-select: none` CSS avoids the context menu for you on your device. [Mushrum Cards](https://github.com/piitaya/lovelace-mushroom) uses this technique to block the context menu and avoids at the same time text selection, and it works on Chrome, Safari and Firefox on desktop devices, but it would be good to know if this works also on your specific device with those capabilities of opening the context menu on long press.
Regards

**Comment by elchininet** [comment]
@ebroder,
Anyway, apart from your specific tests, I‘ll implement it when I get some time. Some kiosk devices run in environments with mouse devices and as the `block_mouse` option it will be good to prevent the content menu appearing.
Regards

**Comment by ebroder** [comment]
I managed to get the Chrome DevTools wired up to my kiosk to do some testing. I was able to confirm that `document.addEventListener('contextmenu', e => e.preventDefault())` does seem to work to prevent the context menu from appearing.

I wasn't able to get either of your alternatives to work. Turning on `block_mouse` mode does (as expected) prevent any tap-based interactions, but tap-and-hold still opens the context menu. I also tried adding `user-select: none` to the `body::after` element that is added by `block_mouse`. That doesn't seem to prevent the context menu either.

I started by opening an issue to see if you were open to this feature. Since you are, I can take a stab at implementing it. I'll try to send you a PR today or tomorrow.
